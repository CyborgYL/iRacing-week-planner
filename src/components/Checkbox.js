// @flow

import * as React from 'react';
import classnames from 'classnames';

import styles from '../styles/main.scss';

type Props = {
  id: string,
  onChange?: (newValue: boolean) => void,
  checked?: boolean,
  disabled?: boolean,
  children?: React.Node,
};

export default function Checkbox({ id, onChange, checked, disabled, children }: Props): React.Node {
  return (
    <div className={classnames({ [styles.checkbox]: true, [styles.disabled]: disabled })}>
      <label htmlFor={id}>
        <input
          id={id}
          type="checkbox"
          onChange={(e) => (onChange(e.target.checked))}
          checked={checked}
          disabled={disabled}
        />
        {children}
      </label>
    </div>
  );
}

Checkbox.defaultProps = {
  checked: false,
  disabled: false,
  onChange: () => {},
  children: null,
};
