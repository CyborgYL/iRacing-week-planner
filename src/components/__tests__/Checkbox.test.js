import { shallow } from 'enzyme';
import React from 'react';
import { describe, test } from '@jest/globals';

import Checkbox from '../Checkbox';

describe('components/Checkbox', () => {
  test('renders', () => {
    const onChange = jest.fn();
    const render = shallow(<Checkbox onChange={onChange}>Test</Checkbox>);

    expect(render).toMatchSnapshot();

    render.find('input').at(0).simulate('change', { target: { checked: true }});
    expect(onChange).toHaveBeenCalled();
  });

  test('renders disabled', () => {
    const onChange = jest.fn()
    const render = shallow(<Checkbox disabled onChange={onChange} />);

    expect(render).toMatchSnapshot();

    render.find('input').at(0).simulate('click');
    expect(onChange).not.toHaveBeenCalled();
  });

  test('renders checked', () => {
    const render = shallow(<Checkbox checked />);

    render.find('input').at(0).simulate('change', { target: { checked: true }});

    expect(render).toMatchSnapshot();
  });
});
