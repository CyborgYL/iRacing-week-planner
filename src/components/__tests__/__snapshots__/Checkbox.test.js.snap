// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Checkbox renders 1`] = `
<div
  className="checkbox"
>
  <label>
    <input
      checked={false}
      disabled={false}
      onChange={[Function]}
      type="checkbox"
    />
    Test
  </label>
</div>
`;

exports[`components/Checkbox renders checked 1`] = `
<div
  className="checkbox"
>
  <label>
    <input
      checked={true}
      disabled={false}
      onChange={[Function]}
      type="checkbox"
    />
  </label>
</div>
`;

exports[`components/Checkbox renders disabled 1`] = `
<div
  className="checkbox disabled"
>
  <label>
    <input
      checked={false}
      disabled={true}
      onChange={[Function]}
      type="checkbox"
    />
  </label>
</div>
`;
